vars/getFunctionalTestStage.groovy

```diff
@@ -61,6 +62,9 @@ Map call(Map kwargs = [:]) {
                 node(cachedCommitPragma("Test-label${pragma_suffix}", label)) {
                     // Ensure access to any branch provisioning scripts exist
                     echo "[${name}] Check out '${base_branch}' from version control"
+                    if (!base_branch) {
+                        error 'base_branch argument is "' + base_branch + '"'
+                    }
```

Comment:
This is true.  I'm not sure why the allowance for an empty `base_branch` was made on L74-L76.  When it is empty, the job fails due to not checking out the branch that the (i.e. daily, weekly) testing branch wants to test.

Maybe @phender can shed some light?

Reply:
Ultimately, the problem here is a(n apparent) Jenkins bug wherein a parameter's default value is not being used and instead the parameter is set to empty (`null`) so the expected branch (that was the default value of the parameter) is not actually getting checked out.

So while I'm not sure what @phender's expected behaviour here was when the `base_branch` was intentionally left empty, this Jenkins bug seems to prevent us from being able to effect that.

If his use case for an empty `base_branch` is not actually being used anywhere, perhaps we can go ahead with this as it is and I can remove the `else` branch in a future PR, given how much CI (and wall clock) time this PR has chewed up so far.

Reply:
I'm okay with that as long as we don't forget about it :)

Reply:
@phender Can you please comment on this bit of code and it's intention?  My addition of the `error()` here seems to be causing test failures:  https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-pipeline-lib-PR-431-master/6/pipeline/

But this `error()` is needed because it seems that this parameter can be empty even though it's being set with a default in the case of the weekly testing branch(es).